Add function implementation for checking mx record + clang format fix

irog-c opened the pull request:
Done in PR:
- Added a function `check_mx` to perform email validation using the SMTP `RCPT TO` command.
- Established a TCP connection to the mail server using Boost.Asio.
- Implemented error handling for SMTP responses
- Fixed small issue with clang format

Testing:
- Currently no way to test this since I don't have the port 25 forwarded on my machine
- Also no way of making unit tests currently (unless I use mocks but not sure what to mock here exactly, we can discuss this together)

irog-c committed:
Add function implementation for checking mx record + clang format fix

KyryloKupin commented on source/EmailVerification.cpp:
```diff
@@ -199,4 +204,47 @@ auto tuposoft::vrf::operator<<(std::ostream &os, const vrf_data &data) -> declty
+
+    while (error && endpoint_iterator != asio::ip::tcp::resolver::iterator()) {
+        socket.close();
+        socket.connect(*endpoint_iterator++, error);
```
> Clang-Tidy: The value returned by this function should not be disregarded; neglecting it may lead to errors

I have clang-tidy in CLion out of the box, but we can add it to the repo. Check [the starter](https://github.com/TheLartians/ModernCppStarter/blob/master/cmake/tools.cmake).

irog-c replied:
[ASIO documentation](https://live.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/basic_stream_socket/connect/overload1.html) says this function returns `void`, not sure what I should do here.

KyryloKupin replied:
It shows me this signature:
![image](https://github.com/TupoSoft/vrf/assets/95937194/e262145c-1385-4891-942e-1bde082a44ef)

KyryloKupin commented on source/EmailVerification.cpp:
```diff
@@ -199,4 +204,47 @@ auto tuposoft::vrf::operator<<(std::ostream &os, const vrf_data &data) -> declty
+    std::string server_response;
+    std::getline(response_stream, server_response);
+
+    return std::atoi(server_response.substr(0, 3).c_str());
```
> Clang-Tidy: 'atoi' used to convert a string to an integer value, but function will not report conversion errors; consider using 'strtol' instead

irog-c replied:
I used `stoi` which works with `std::string`.

irog-c committed:
Change request

irog-c committed:
Tests

irog-c committed:
Check error code during socket connect

KyryloKupin commented on test/CMakeLists.txt:
```diff
@@ -9,28 +9,38 @@ project(${PROJECT_NAME}_test VERSION ${PROJECT_VERSION})
+CPMAddPackage(
+  NAME googletest
+  GITHUB_REPOSITORY google/googletest
+  GIT_TAG release-1.12.1
```
Why old version?

KyryloKupin committed:
Refactor email verification system for better testing capabilities

This commit modifies the email verification system by dividing some functionalities into separate components to facilitate better unit testing. This involves restructuring code, renaming classes, creating new socket wrapper files, extracting TCP socket related code to a socket wrapper, and updating associated tests accordingly. Some obsolete tests and configurations have also been removed from the system. Additional changes include switching to `find_package` for installing packages, enforcing standard conformance on MSVC, and updating google test version.
